lib/src/extensions.dart

```diff
@@ -28,8 +28,16 @@ extension DateTimeTimeExtension on DateTime {
   /// Adds this DateTime and Duration and returns the sum as a new DateTime object.
   DateTime operator +(Duration duration) => add(duration);
 
-  /// Subtracts the Duration from this DateTime returns the difference as a new DateTime object.
-  DateTime operator -(Duration duration) => subtract(duration);
+  /// If used with a Duration, it subtracts the Duration from this DateTime returns the difference as a new DateTime object.
+  /// If used with a DateTime it returns the difference between the two DateTime objects as a Duration.
```

Comment:
I don't believe we can actually have this. There is currently no language support for unions types. We might just have to wait this one out.

Reply:
We also cannot add overloads on methods (`subtract`). Unless we create an entirely new method for this. Say `subtractDateTime` or `minus`.